### PR TITLE
Add multisig static example

### PR DESCRIPTION
**Pull request by lexnv**
This example executes a `multisig::as_multi` extrinsic
containing a `Assets:freeze_asset` call.

While at it, also add the substrate scale metadata to facilitate
working with the substrate nodes.

Part of https://github.com/paritytech/subxt/issues/712.

### CODE REVIEW

File: examples/examples/multisig.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! To run this example, a local polkadot node should be running. Example verified against substrate latest-38a955ba4a0.
+//!
+//! E.g.
+//! ```bash
+//! curl "https://releases.parity.io/substrate/x86_64-debian:stretch/latest/substrate/substrate" --output /usr/local/bin/substrate --location
+//! substrate --dev --tmp
+//! ```
```
Comment (jsdw) [inline examples/examples/multisig.rs:11]: Is this the only example using Substrate? I think just for consistency with other examples maybe it's best to stick to Polkadot?

Comment (lexnv) [reply examples/examples/multisig.rs:11]: The Polkadot node does not contain the `Assets` pallet, at least I didn't see it in the metadata of `wss://rpc.polkadot.io:443`

Comment (jsdw) [reply examples/examples/multisig.rs:11]: Oh, that's ok; we don't need to specifically use this Assets call as the example; in https://github.com/paritytech/subxt/pull/714/files I used a balance transfer so perhaps we could standardise and both just do that?

Comment (jsdw) [reply examples/examples/multisig.rs:11]: Also I think I'll rename my example to `dynamic_multisig.rs` to align with yours :)

File: examples/examples/multisig.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    // Create a client to use:
+    let api = OnlineClient::<SubstrateConfig>::new().await?;
+
+    // The call data for the `Multisig::as_multi` extrinsic.
+    // This must be of type `RuntimeCall` and represents the `Assets::freeze_asset` method.
+    // Otherwise, could have been obtained using `substrate::tx().assets().freeze_asset(100)`
+    let call = substrate::runtime_types::kitchensink_runtime::RuntimeCall::Assets(
+        substrate::runtime_types::pallet_assets::pallet::Call::freeze_asset { id: 100 },
+    );
```
Comment (jsdw) [inline examples/examples/multisig.rs:37]: what about:

```rust
substrate::tx().assets().freeze_asset(100).call_data().clone()
```

does that work instead?

And if so, perhaps we could add an `into_call_data()` to mirror the `into_value() I added on the dynamic side, and avoid the clone here?

**Comment by jsdw** [comment]
Since Alex is away today I'll combine this with my PR and fix anything up there to make the two consistent :)

**Pull request closed without merging**